lib/nfg-evo-rest-client/base.rb

```diff
@@ -33,7 +33,8 @@ def self.append_param_fields_if_any(request)
       # You can set the request objects values by passing in a hash of
       # values when instantiating the object.
       # i.e. NfgEvoRestClient::Donation.new(from_date_time: "2016-02-01", to_date_time: "2016-02-29")
-      param_fields.each do |field_or_hash|
+      fields = request.object.respond_to?(:param_fields) ? request.object.param_fields : param_fields
```

Comment:
We've defined param_fields as instance methods in all derived classes, and in the base class, we've added it as a class method. However, when we call param_fields from here, it always invokes the method from the base class. that is is why customs params were never appended to URL(except on event.rb where we have added HACK),